samples/net/cloud/google_iot_mqtt/src/protocol.c

```diff
@@ -166,7 +166,7 @@ void mqtt_evt_handler(struct mqtt_client *const client,
 
 		/* assuming the config message is textual */
 		while (len) {
-			bytes_read = mqtt_read_publish_payload(
+			bytes_read = mqtt_read_publish_payload_blocking(
 				client, d,
 				len >= 32 ? 32 : len);
 			if (bytes_read < 0 && bytes_read != -EAGAIN) {
```

Comment:
I think we don't need to check for `bytes_read != -EAGAIN` here either, just break on any error.